win32ss/user/ntuser/ime.c

```diff
@@ -80,14 +80,72 @@ NtUserCheckImeHotKey(
     return 0;
 }
 
-
-DWORD
+BOOL
 APIENTRY
 NtUserDisableThreadIme(
-    DWORD dwUnknown1)
+    DWORD dwThreadID)
 {
-    STUB;
-    return 0;
+    PTHREADINFO pti, ptiCurrent;
+    PPROCESSINFO ppi;
+    BOOL ret = FALSE;
+
+    UserEnterExclusive();
+
+    if (!IS_IMM_MODE())
+    {
+        EngSetLastError(ERROR_CALL_NOT_IMPLEMENTED);
+        goto Quit;
+    }
+
+    ptiCurrent = GetW32ThreadInfo();
+
+    if (dwThreadID == (DWORD)-1)
+    {
+        ppi = ptiCurrent->ppi;
+        ppi->W32PF_flags |= W32PF_DISABLEIME;
+
+        for (pti = ppi->ptiList; pti; )
+        {
+            pti->TIF_flags |= TIF_DISABLEIME;
+
+            if (pti->spwndDefaultIme)
+            {
+                co_UserDestroyWindow(pti->spwndDefaultIme);
+                pti->spwndDefaultIme = NULL;
+                pti = ptiCurrent->ppi->ptiList;
```

Comment:
looks like it's equivalent to:
```suggestion
                pti = ppi->ptiList;
```
(is it true that neither `ptiCurrent` nor `ppi` seems to change here? or not?)
But if this is so, then does this mean you restart looping from the beginning of the list?
If any case this definitively need a comment.

Reply:
Simplified.